presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java

```diff
@@ -239,13 +239,22 @@ public LateralJoinNode lateral(List<Symbol> correlation, PlanNode input, PlanNod
 
     public TableScanNode tableScan(List<Symbol> symbols, Map<Symbol, ColumnHandle> assignments)
     {
+        return tableScan(symbols, assignments, null);
+    }
+
+    public TableScanNode tableScan(List<Symbol> symbols, Map<Symbol, ColumnHandle> assignments, Expression originalConstraint)
+    {
         TableHandle tableHandle = new TableHandle(new ConnectorId("testConnector"), new TestingTableHandle());
-        return tableScan(tableHandle, symbols, assignments);
+        return tableScan(tableHandle, symbols, assignments, originalConstraint);
     }
 
     public TableScanNode tableScan(TableHandle tableHandle, List<Symbol> symbols, Map<Symbol, ColumnHandle> assignments)
     {
-        Expression originalConstraint = null;
+        return tableScan(tableHandle, symbols, assignments, null);
+    }
+
+    public TableScanNode tableScan(TableHandle tableHandle, List<Symbol> symbols, Map<Symbol, ColumnHandle> assignments, Expression originalConstraint)
```

Comment:
will change the commit message. 

> Oh, and the originalConstraint field bears the following comment, are you sure you want to set it here?

I would not if I could not. That comment is outdated BTW, as it also says:

>    // To make this work, the originalConstraint should be set exactly once after the first predicate push down and never adjusted after that.
    // In this way, we are always guaranteed to have a readable predicate that provides some kind of upper bound on the constraints.
 
as there are rules that change the constraint (simplify it).